src/nested_pandas/series/accessor.py

```diff
@@ -99,7 +99,8 @@ def to_flat(self, fields: list[str] | None = None) -> pd.DataFrame:
                 name=field,
                 copy=False,
             )
-        return pd.DataFrame(flat_series)
+        # Note that index should now not be none since we iterated over at least one field
+        return pd.DataFrame(flat_series, index=pd.Series(index, name=self._series.index.name))
```

Comment:
Could we put it to line 95 instead?

Reply:
I mean I believe that it would be enough to put the index into each pd.Series instead of putting it into result dataframe

Reply:
Yeah, that's reasonable. Done.